src/components/core/ApplicationNavigationBar.tsx

```diff
@@ -1,25 +1,82 @@
-import { Navbar, Text } from "@mantine/core";
+import React, { createContext, useContext, useState } from "react";
+import { Navbar, List } from "@mantine/core";
+import Link from "next/link";
+import { navigationItems, FaChevronDown } from "./ApplicationHeader";
 
 interface ApplicationNavigationBarProps {
-	opened: boolean;
+  opened: boolean;
+  onClose: () => void; 
 }
 
-export default function ApplicationNavigationBar({ opened }: ApplicationNavigationBarProps) {
-	return (
-		<Navbar
-			p="md"
-			hiddenBreakpoint={Number.MAX_VALUE} // XD
-			hidden={!opened}
-			width={
-				opened
-					? { sm: 200, lg: 300 }
-					: {
-							sm: 0,
-							lg: 0,
-					  }
-			}
-		>
-			<Text>Application navbar</Text>
-		</Navbar>
-	);
+export default function ApplicationNavigationBar({
+  opened,
+  onClose, 
+}: ApplicationNavigationBarProps) {
+  const [activeIndex, setActiveIndex] = useState<number | null>(null);
+
+  const handleItemClick = (index: number) => {
+    if (index === activeIndex) {
+      setActiveIndex(null); // Toggle off if already active
+    } else {
+      setActiveIndex(index); // Activate index and show subitems
+    }
+    onClose(); 
+  };
+
+  return (
+    <Navbar
+      className="m-nav min-w-max"
+      p="md"
+      hiddenBreakpoint={Number.MAX_VALUE}
+      hidden={!opened}
+      width={
+        opened
+          ? { sm: 200, lg: 300 }
+          : {
+              sm: 0,
+              lg: 0,
+            }
+      }
+    >
+      {/* main-div */}
+      <div className="flex w-max w-auto">
+        {/* list container */}
+        <div className="flex w-full min-w-max justify-center">
+          {/* list-box */}
+          <List className="list-none">
+            {navigationItems.map((item, index) => (
+              <List.Item key={index} className="m-7 text-3xl">
+                <div
+                  className="flex cursor-pointer items-center"
+                  onClick={() => handleItemClick(index)}
+                >
+                  <Link href={item.href} className="text-white no-underline">
+                    {item.title}
+                  </Link>
+                  {item.subitems && (
+                    <FaChevronDown
+                      className={`ml-3 transform ${
+                        activeIndex === index ? "rotate-180" : "rotate-0"
+                      } transition-transform duration-200 ease-in-out`}
+                    />
+                  )}
+                </div>
+                {item.subitems && activeIndex === index && (
+                  <List className="mt-5">
+                    {item.subitems.map((subitem, subindex) => (
+                      <List.Item key={subindex}>
+                        <Link href="" className="text-2xl text-white no-underline">
```

Comment:
pls also make sure the href on Links are set, in this case, it's a subitem so check if the subitem is a link or not